diff --git a/drivers/input/touchscreen/goodix_9916r/goodix_ts_core.c b/drivers/input/touchscreen/goodix_9916r/goodix_ts_core.c
index 64c82e6e273d..a1a3ded40ebd 100644
--- a/drivers/input/touchscreen/goodix_9916r/goodix_ts_core.c
+++ b/drivers/input/touchscreen/goodix_9916r/goodix_ts_core.c
@@ -888,18 +888,6 @@ static ssize_t goodix_ts_report_rate_store(struct device *dev,
 					struct device_attribute *attr,
 					const char *buf, size_t count)
 {
-	struct goodix_ts_core *core_data = dev_get_drvdata(dev);
-
-	if (!buf || count <= 0)
-		return -EINVAL;
-
-	if (buf[0] != '0') {
-		goodix_core_data->report_rate = 480;
-		core_data->hw_ops->switch_report_rate(core_data, true);
-	} else {
-		goodix_core_data->report_rate = 240;
-		core_data->hw_ops->switch_report_rate(core_data, false);
-	}
 	return count;
 }
 
@@ -2641,6 +2629,12 @@ static int goodix_later_init_thread(void *data)
 		ts_err("stage2 init failed");
 		goto uninit_fw;
 	}
+
+	if (cd->report_rate != 240) {
+		ts_info("%s: Enable high sampling rate", __func__);
+		hw_ops->switch_report_rate(cd, true);
+	}
+
 	cd->init_stage = CORE_INIT_STAGE2;
 
 	return 0;
@@ -3015,19 +3009,6 @@ static void goodix_set_game_work(struct work_struct *work)
 	if (ret < 0)
 		ts_err("send game mode fail");
 
-	if (xiaomi_touch_interfaces.touch_mode[Touch_Tolerance][SET_CUR_VALUE] >
-	    xiaomi_touch_interfaces.touch_mode[Touch_Tolerance][GET_DEF_VALUE]) {
-		if (goodix_core_data->report_rate == 240) {
-			hw_ops->switch_report_rate(goodix_core_data, true);
-			goodix_core_data->report_rate = 480;
-		}
-	} else {
-		if (goodix_core_data->report_rate == 480) {
-			hw_ops->switch_report_rate(goodix_core_data, false);
-			goodix_core_data->report_rate = 240;
-		}
-	}
-
 	mutex_unlock(&goodix_core_data->core_mutex);
 }
 
@@ -3659,7 +3640,7 @@ static int goodix_ts_probe(struct platform_device *pdev)
 	device_init_wakeup(&pdev->dev, 1);
 	core_data->init_stage = CORE_INIT_STAGE1;
 	core_data->charger_status = -1;
-	core_data->report_rate = 240;
+	core_data->report_rate = 480;
 	goodix_modules.core_data = core_data;
 	core_module_prob_sate = CORE_MODULE_PROB_SUCCESS;
 
